### PR TITLE
Update the canape module

### PR DESCRIPTION
**Pull request by PouuleT**
Wishlists are now aquired via two different calls for shows and movies
in the new canape version

### CODE REVIEW

File: modules/canape/canape.go [hunk 1]
```diff
@@ -97,55 +141,59 @@ func (w *Wishlist) getUsersWishlists() (*polochon.Wishlist, error) {
...
+	req, err := http.NewRequest("GET", fmt.Sprintf("%s/%s", w.URL, URL), nil)
 	if err != nil {
-		return nil, err
+		return err
 	}
 
-	response := &response{}
-	err = json.Unmarshal(body, response)
+	// Add the auth headers
+	req.Header.Add("Authorization", w.Token)
```
Comment (gregdel) [inline modules/canape/canape.go]: Shouldn't this be something like: 'Authorization: Bearer token_here' to be jwt compliant ?

File: modules/canape/canape.go [hunk 2]
```diff
@@ -97,55 +141,59 @@ func (w *Wishlist) getUsersWishlists() (*polochon.Wishlist, error) {
...
 	if err != nil {
 		return nil, err
 	}
-	defer resp.Body.Close()
+	return wishlist.Movies, nil
 
-	body, err := ioutil.ReadAll(resp.Body)
+}
+
+func (w *UserWishlist) request(URL, response interface{}) error {
```
Comment (gregdel) [inline modules/canape/canape.go]: We might want URL to be a string instead of a an interface

File: modules/canape/canape.go [hunk 3]
```diff
@@ -97,55 +141,59 @@ func (w *Wishlist) getUsersWishlists() (*polochon.Wishlist, error) {
...
+	resp, err := client.Do(req)
 	if err != nil {
-		return nil, err
+		return err
 	}
+	defer resp.Body.Close()
 
-	return response, nil
-}
+	if resp.StatusCode != 200 {
```
Comment (gregdel) [inline modules/canape/canape.go]: http.StatusOK

File: modules/canape/canape.go [hunk 4]
```diff
@@ -97,55 +141,59 @@ func (w *Wishlist) getUsersWishlists() (*polochon.Wishlist, error) {
...
+		return fmt.Errorf("canape: invalid http code %q", resp.Status)
+	}
 
-// GetMovieWishlist gets the movies wishlist
-func (w *Wishlist) GetMovieWishlist(log *logrus.Entry) ([]*polochon.WishedMovie, error) {
-	wl, err := w.getUsersWishlists()
+	body, err := ioutil.ReadAll(resp.Body)
```
Comment (gregdel) [inline modules/canape/canape.go]: Are you sure we need to read the whole body ?

File: modules/canape/canape.go (hunk 2 above)
Comment (PouuleT) [inline modules/canape/canape.go]: Yup

File: modules/canape/canape.go (hunk 1 above)
Comment (PouuleT) [inline modules/canape/canape.go]: Why not!

File: modules/canape/canape.go (hunk 3 above)
Comment (PouuleT) [inline modules/canape/canape.go]: Done!

File: modules/canape/canape.go (hunk 4 above)
Comment (PouuleT) [inline modules/canape/canape.go]: nope

**Pull request merged**